### ISSUE TITLE
Render valid optimized AMP by default

### ISSUE THREAD

**Issue by sebastianbenz** [zeit/next.js#7461]
# Feature request

AMP Optimizer can now produce valid optimized AMP and it's no longer necessary to serve (previously invalid) optimized AMP in parallel with a valid AMP version 🎉. You can see it in action on https://amp.dev. 

This means Next.js can now serve valid optimized AMP pages for both AMP-only and hybrid AMP mode:

* Hybrid AMP (pages/about.js) would output:

  out/about/index.html - with client-side React runtime
  **out/about.amp/index.html - valid optimized AMP page (to optimize for platforms linking to AMP pages on the origin, e.g. twitter)**

* AMP-only (pages/about.js) would output:

  **out/about/index.html - valid optimized AMP page**
~~out/about.amp/index.html - AMP page~~

## What's needed

Valid optimized AMP support will be officially launched in a new package under the @ampproject scope (there are a few breaking API changes, which makes it a good time for us to switch to the official package scope). 

Until then, you can use the [pre-release version](https://www.npmjs.com/package/amp-toolbox-optimizer/v/1.2.0-alpha.1) which already includes the final API.

Things that need to be done:

- [ ] upgrade to amp-toolbox-optimizer@1.2.0-alpha.1
- [ ] migrate [optimize.ts](packages/next-server/server/optimize-amp.ts) to the new optimizer API

  ```
  const html = '...';
  const AmpOptimizer = require('amp-toolbox-optimizer');
  const optimizer = AmpOptimizer.create(); // no additional config required for valid AMP mode
  const transformedHtml = await optimizer.transformHtml(html);
  ```

  The `{ amphtml, query }` options are no longer needed.
- [ ] [render.tsx](packages/next-server/server/render.rsx) can be simplified to: 

  ```
  if (amphtml && html) {
    // use replace to allow rendering directly to body in AMP mode
    html = html.replace('__NEXT_AMP_RENDER_TARGET__', docProps.html)

    html = await optimizeAmp(html)

    if (renderOpts.ampValidator) {
      await renderOpts.ampValidator(html, pathname)
    }
  }
  ``` 

I hope I got everything...

## Future changes

* AMP Optimizer currently sets `<html transformed=google;v1 ...`, however subsequent AMP Optimizer updates will change this to  `<html transformed=self ...`.
* The AMPHTML Validator will support self-hosted the AMP runtime and AMP extensions in the future. This means this will become valid AMP:

  ```
  <html transformed=self>
  <head>
    <script async src="amp-runtime-v0.js"></script>
  </head>
  ...
  </html>
  ```

//cc @timneutkens @ijjk

**Comment by MarchWorks** [comment]
@ijjk what is the difference between `out/about.amp/index.html ` in the hybrid mode and  `out/about/index.html` in the AMP-only mode?

**Comment by skptricks** [comment]
AMP feature in next.js is not working. I have tried below source code :

import 'isomorphic-fetch'
import React from 'react'
import Link from 'next/link'

import { useAmp } from 'next/amp'
export const config = { amp: true };

const app = ({ stars }) => (

dddddddddddddssssssssssssssssssssss

First Post
) export default app
Error Message : ---
Cannot read property '__html' of undefined
TypeError: Cannot read property '__html' of undefined
at D:\sumitprojects\NextSimpleStarter.next\server\static\development\pages_document.js:819:76
at Array.map ()
at Head.render (D:\sumitprojects\NextSimpleStarter.next\server\static\development\pages_document.js:819:27)
at processChild (D:\sumitprojects\NextSimpleStarter\node_modules\react-dom\cjs\react-dom-server.node.development.js:3171:18)
at resolve (D:\sumitprojects\NextSimpleStarter\node_modules\react-dom\cjs\react-dom-server.node.development.js:3013:5)
at ReactDOMServerRenderer.render (D:\sumitprojects\NextSimpleStarter\node_modules\react-dom\cjs\react-dom-server.node.development.js:3436:22)
at ReactDOMServerRenderer.read (D:\sumitprojects\NextSimpleStarter\node_modules\react-dom\cjs\react-dom-server.node.development.js:3395:29)
at Object.renderToStaticMarkup (D:\sumitprojects\NextSimpleStarter\node_modules\react-dom\cjs\react-dom-server.node.development.js:3969:27)
at renderDocument (D:\sumitprojects\NextSimpleStarter\node_modules\next-server\dist\server\render.js:89:18)
at Object.renderToHTML (D:\sumitprojects\NextSimpleStarter\node_modules\next-server\dist\server\render.js:299:16)

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
